doc/high-frequency-telemetry/high-frequency-telemetry-hld.md

```diff
@@ -0,0 +1,714 @@
+      }
+    ]
+  }
+}
+```
+
+For design goals, requirements, and specification of the OTLP, please refer to the official documentation: [OpenTelemetry Protocol (OTLP)](https://github.com/open-telemetry/opentelemetry-proto/tree/main/docs).
+
+For practical OTLP message examples and implementation patterns, see the examples in the OpenTelemetry repository: [OpenTelemetry Protocol Examples](https://github.com/open-telemetry/opentelemetry-proto/tree/main/examples).
+
+### 7.3. Bandwidth Estimation
+
+We estimate the bandwidth based only on the effective data size, not the actual data size. The extra information in a message, such as the IPFIX header (16 bytes), data prefix (4 bytes), and observation time nanoseconds (8 bytes), is negligible. For example, a IPFIX message could include $The Maximal Number Of Counters In One Message = \frac{0xFFFF_{Max Length Bytes} - 16_{Header Bytes} - 4_{DataPrefix Bytes} - 8_{Observation Time Nanoseconds Bytes}}{8_{bytes}} \approx 8188$, So $The Percentage Of Effective Data = \frac{0xFFFF_{Max Length Bytes} - 16_{Header Bytes} - 4_{DataPrefix Bytes} - 8_{Observation Time Nanoseconds Bytes}} {0xFFFF_{Max LengthBytes}} \approx 99.9\%$ .
+
+The following table is an example of telemetry bandwidth of one cluster
+
+| # of stats per port | # of ports per switch | # of switch | frequency (us) | Total BW per switch(Mbps) | Total BW(Mbps) |
+| ------------------- | --------------------- | ----------- | -------------- | ------------------------- | -------------- |
+| 30                  | 64                    | 10,000      | 10             | 12,288                    | 122,880,000    |
+
+- ${Total BW Per Switch} = \frac{{\verb|#| Of Stats Per Port} \times 8_{bytes} \times {\verb|#| Of Ports Per Switch} \times {Frequency} \times 1,000 \times 8}{1,000,000}$
+- ${Total BM} = {Total BW Per Switch} \times {\verb|#| Of Switch}$
+
+### 7.4. Config DB
+
+Any configuration changes in the config DB will interrupt the existing session and initiate a new one.
+
+#### 7.4.1. DEVICE_METADATA
+
+```
+DEVICE_METADATA|localhost
+    "high_frequency_telemetry_chunk_size": {{uint32}}
+    "high_frequency_telemetry_chunk_count": {{uint32}} (Optional)
+```
+
+```
+; field                      = value
+high_frequency_telemetry_chunk_size  = uint32; reporting byte size of chunk under the high frequency telemetry.
+high_frequency_telemetry_chunk_count = uint32; chunk count under the high frequency telemetry. Some platforms may not support setting this value.
+```
+
+#### 7.4.2. HIGH_FREQUENCY_TELEMETRY_PROFILE
+
+```
+HIGH_FREQUENCY_TELEMETRY_PROFILE|{{profile_name}}
+    "stream_state": {{enabled/disabled}}
+    "poll_interval": {{uint32}}
+```
+
+```
+key                = HIGH_FREQUENCY_TELEMETRY_PROFILE|profile_name a string as the identifier of high frequency telemetry
+; field            = value
+stream_state       = enabled/disabled ; Enabled/Disabled stream.
+poll_interval      = uint32 ; The interval to poll counter, unit microseconds.
```

Comment:
hi Ze, should we add a remote write endpoint in this profile?

Reply:
What kind of endpoint? for OpenTelemetry collector or outside service like InflexDB?

Reply:
Done， please check it